tests/test_utils/test_stringify.py

```diff
@@ -0,0 +1,327 @@
+        )
+        self.assertEqual(
+            stringify_span_group(span_group=query_span_group, document=doc),
+            "This is a test.",
+        )
+
+    def test_partial_word_match(self):
+        doc = Document.from_json(
+            {
+                "symbols": "This is a test.",
+                "words": [
+                    {"id": 0, "spans": [{"start": 0, "end": 4}], "text": "This"},
+                    {"id": 1, "spans": [{"start": 5, "end": 7}], "text": "is"},
+                    {"id": 2, "spans": [{"start": 8, "end": 9}], "text": "a"},
+                    {"id": 3, "spans": [{"start": 10, "end": 14}], "text": "test"},
+                    {"id": 4, "spans": [{"start": 14, "end": 15}], "text": "."},
+                ],
+            }
+        )
+        # make sure test fixture is defined correctly
+        for word in doc.words:
+            assert word.text == doc.symbols[word.start : word.end]
+        
+        # does it grab partial word matches?
+        query_span_group = SpanGroup.from_json(
+            {
+                "spans": [
+                    {"start": 2, "end": 7},
+                ]
+            }
+        )
+        self.assertEqual(
+            stringify_span_group(span_group=query_span_group, document=doc),
+            "This is",
+        )
+
+         query_span_group = SpanGroup.from_json(
+            {
+                "spans": [
+                    {"start": 6, "end": 13},
+                ]
+            }
+        )
+        self.assertEqual(
+            stringify_span_group(span_group=query_span_group, document=doc),
+            "is a test",
+        )
+
+    def test_use_word_metadata_text(self):
+        doc = Document.from_json(
+            {
+                "symbols": "This is a te-\nst.",
+                "words": [
+                    {"id": 0, "spans": [{"start": 0, "end": 4}], "text": "This"},
+                    {"id": 1, "spans": [{"start": 5, "end": 7}], "text": "is"},
+                    {"id": 2, "spans": [{"start": 8, "end": 9}], "text": "a"},
+                    {"id": 3, "spans": [{"start": 10, "end": 16}], "text": "test"},
+                    {"id": 4, "spans": [{"start": 16, "end": 17}], "text": "."},
+                ],
+            }
+        )
+        # make sure test fixture is defined correctly
+        for i, word in enumerate(doc.words):
+            if i != 3:
+                assert word.text == doc.symbols[word.start : word.end]
+            else:
+                assert word.text == "test"
+                assert doc.symbols[word.start : word.end] == "te-\nst"
+        
+        query_span_group = SpanGroup.from_json(
+            {
+                "spans": [
+                    {"start": 5, "end": 17},
+                ]
+            }
+        )
+        self.assertEqual(
+            stringify_span_group(span_group=query_span_group, document=doc),
+            "is a test.",
+        )
+
+    def test_how_words_relate_to_stringify(self):
```

Comment:
@regan-huff @cmwilhelm does this test help explain the confusing behavior about `words` and `stringify()`?